lib/mongo/grid/fs_bucket.rb

```diff
@@ -487,8 +487,26 @@ def files_name
 
       def ensure_indexes!
         if files_collection.find({}, limit: 1, projection: { _id: 1 }).first.nil?
-          chunks_collection.indexes.create_one(FSBucket::CHUNKS_INDEX, :unique => true)
-          files_collection.indexes.create_one(FSBucket::FILES_INDEX)
+          create_index_if_missing!(files_name, FSBucket::FILES_INDEX)
+        end
+
+        if chunks_collection.find({}, limit: 1, projection: { _id: 1 }).first.nil?
+          create_index_if_missing!(chunks_name, FSBucket::CHUNKS_INDEX, :unique => true)
+        end
+      end
+
+      def create_index_if_missing!(collection_name, index_spec, options = {})
+        begin
+          unless database[collection_name].indexes.get(index_spec).nil?
+            database[collection_name].indexes.create_one(index_spec, options)
```

Comment:
This appears to create the index if it exists?

Reply:
🤦🏻

Reply:
Fixed my backwards logic (finally)